File: Backend/manga-world/src/main/java/com/amakakeru/mangaworld/Controller/CategoryController.java
package com.amakakeru.mangaworld.Controller;


import com.amakakeru.mangaworld.Entity.Category;
import com.amakakeru.mangaworld.Repository.CategoryRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@CrossOrigin(origins = "http://localhost:3000")
@RestController
public class CategoryController {

    @Autowired
    private CategoryRepository categoryRepository;

    @GetMapping("/allCategories")
    public List<Category> getMangaByCategory() {
//        select c1_0.c_id,c1_0.c_name from category c1_0
        return categoryRepository.findAll();
    }

    @PostMapping("/categories/addNewCategory")
    private boolean addNewCategory(@RequestBody Category category) {
        Category category1 = categoryRepository.findCategoryByCname(category.getCname());
        if (category1 == null) {
            categoryRepository.save(category);
            return true;
        } else {
            return false;
        }
    }

}

File: Frontend/manga-world/src/Controller/CategoryController.ts
import axios from "axios";

const USER_API_BASE_URL = "http://localhost:8080/";

class CategoryController {
  allCategories() {
    return axios.get(USER_API_BASE_URL + "allCategories");
  }
  addNewCategory(category: any) {
    return axios.post(
      USER_API_BASE_URL + "categories/addNewCategory",
      category
    );
  }
}

export default new CategoryController();
